File: 프로그래머스/3/43162. 네트워크/네트워크.cpp
#include <string>
#include <vector>

using namespace std;

bool vst[201];

void dfs(int cur, int n, vector<vector<int>> computers) {
    for (int i=0;i<computers[cur].size();i++) {
        if (i == cur) continue;
        if (computers[cur][i] == 0) continue;        
        if (vst[i]) continue;
        
        vst[i] = true;
        dfs(i, n, computers);
    }
}

int solution(int n, vector<vector<int>> computers) {
    int answer = 0;
    for (int i=0;i<n;i++) {
        if (vst[i]) continue;
        
        answer++;
        dfs(i, n, computers);
    }
    
    return answer;
}

File: 프로그래머스/3/43162. 네트워크/네트워크.java
class Solution {

    private boolean[] vst = new boolean[201];
    
    public int solution(int n, int[][] computers) {
        int answer = 0;
        
        for (int i=0;i<n;i++) {
            if (vst[i]) continue;
            
            answer++;
            dfs(n, computers, i);
        }
        
        return answer;
    }
    
    private void dfs(int n, int[][] computers, int cur) {
        
        vst[cur] = true;
        
        for (int next = 0; next< n;next++) {
            if (next == cur) continue;
            if (vst[next]) continue;
            if (computers[cur][next] == 0) continue;
            
            dfs(n, computers, next);
        }
    }
}